### PR TITLE
asserts: fetching code should fetch authority-delegation assertions with signing keys as needed

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
based on #11380

File: asserts/database_test.go [hunk 1]
```diff
@@ -655,7 +655,6 @@ func (safs *signAddFindSuite) TestSignDelegation(c *C) {
 
 	// now add authority-delegation
 	headers = map[string]interface{}{
-		"type":         "account-key",
```
Comment (stolowski) [inline asserts/database_test.go:658]: These changes got me very confused, but I noticed there is a separate commit for them and the commit message clarifies it, so 👍

File: asserts/fetcher.go [hunk 2]
```diff
@@ -106,13 +106,25 @@ func (f *fetcher) Fetch(ref *Ref) error {
...
 		PrimaryKey: []string{keyID},
 	}
-	return f.Fetch(keyRef)
+	if err := f.Fetch(keyRef); err != nil {
+		return err
+	}
+	if signatoryID == authorityID {
+		return nil
+	}
+	delegationRef := &Ref{
```
Comment (stolowski) [inline asserts/fetcher.go:123]: Nitpick, maybe add a clarification comment: `Signatory ID is never empty (it is equal to authorityID if not set)`.

**Review by stolowski — APPROVED** [review]
Looks good.

File: asserts/fetcher.go [hunk 3]
```diff
@@ -106,13 +106,25 @@ func (f *fetcher) Fetch(ref *Ref) error {
...
+		Type:       AuthorityDelegationType,
+		PrimaryKey: []string{authorityID, signatoryID},
+	}
+	return f.Fetch(delegationRef)
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply asserts/fetcher.go:127]: I see, thank you

**Review by MiguelPires — APPROVED** [review]
LGTM

**Pull request merged**